### PR TITLE
fix(iceberg): fix commit_checkpoint_interval for iceberg table with connector

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://raw.githubusercontent.com/risingwavelabs/risingwave/17af8a747593ebdbfa826691daf75bdab7d14fa0/.github/contributor-license-agreement.txt).

## What's changed and what's your intention?

- Previously, we don't remove the `commit_checkpoint_interval` from the connector property which will result in a unknown field.

## Checklist

- [ ] I have written necessary rustdoc comments.
- [ ]  I have added necessary unit tests and integration tests.
- [ ]  I have added test labels as necessary. 
- [ ]  I have added fuzzing tests or opened an issue to track them. 
- [ ]  My PR contains breaking changes. 
- [ ]  My PR changes performance-critical code, so I will run (micro) benchmarks and present the results. 
- [ ]  My PR contains critical fixes that are necessary to be merged into the latest release. 

## Documentation

- [ ]  My PR needs documentation updates. 

<details>
<summary><b>Release note</b></summary>

</details>

### CODE REVIEW

**Review by StrikeW — COMMENTED** [review]
Why not make `commit_checkpoint_interval` to be a known field? I guess just need to add it to the config struct.

**Comment by chenzl25** [comment]
> Why not make `commit_checkpoint_interval` to be a known field? I guess just need to add it to the config struct.

Because I want to make it available in every source connector when creating a table with an iceberg engine, making it a known field will be invasive to every connector property, while it also doesn't make sense to make it a known field when the table is hummock table.

**Review by StrikeW — APPROVED** [review]
Rest lgtm

File: src/frontend/src/handler/create_table.rs [hunk 1]
```diff
@@ -1635,13 +1636,18 @@ pub async fn create_iceberg_engine_table(
         bail!("commit_checkpoint_interval must be a positive integer: 0");
     }
 
+    // remove commit_checkpoint_interval from source options, otherwise it will be considered as an unknown field.
+    source
+        .as_mut()
+        .map(|x| x.with_properties.remove(COMMIT_CHECKPOINT_INTERVAL));
```
[the remark this answers is not in the document]
Comment (chenzl25) [reply src/frontend/src/handler/create_table.rs:1642]: Yes.

**Pull request merged**